packages/discovery-react-components/src/components/StructuredQuery/utils/getFieldNames.ts

```diff
@@ -0,0 +1,16 @@
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+
+export const getFieldNames = (response: DiscoveryV2.ListFieldsResponse | null): string[] => {
+  if (response && response.fields) {
+    let fieldObjects = response.fields;
+    const fieldNames = fieldObjects.reduce((namesArray: string[], fieldObject) => {
+      if (fieldObject.field) {
+        namesArray.push(fieldObject.field);
```

Comment:
since you aren't checking the `collection_id`, you might want to make sure you aren't adding duplicate fields.

i would test this out by creating 2 collections and putting documents in both and then validating that the field names you get back from this function are unique (i don't think they will be)

Reply:
another thing - each field is going to have a `type` -> might want to revisit this later but i don't think all field types would be beneficial to include in this dropdown (namely, we should probably filter out `nested` types)

Reply:
Ah interesting, nice catch. I'll test this out and can also check out what they did for this dropdown in the search results improvement tools section of the Tooling app

Reply:
it was easy enough to filter out fields with type `nested`. I also noticed that in the Tooling repo they were making sure the array of field names were unique, which confirms what you had thought might be the behavior when there are multiple collections with identical field names, so I also utilized `uniq` in our helper before sending this array of field names back to the `RuleRow` component